libwasmvm/src/cache.rs

```diff
@@ -415,6 +416,64 @@ fn do_get_metrics(cache: &mut Cache<GoApi, GoStorage, GoQuerier>) -> Result<Metr
     Ok(cache.metrics().into())
 }
 
+#[derive(Serialize)]
+struct PerModuleMetrics {
+    hits: u32,
+    size: usize,
+}
+
+impl From<cosmwasm_vm::PerModuleMetrics> for PerModuleMetrics {
+    fn from(value: cosmwasm_vm::PerModuleMetrics) -> Self {
+        Self {
+            hits: value.hits,
+            size: value.size,
+        }
+    }
+}
+
+#[derive(Serialize)]
+struct PinnedMetrics {
+    // TODO: Remove the array usage as soon as the metrics have a stable wire format
```

Comment:
Could you elaborate what was meant by this comment?

Given that there are multiple ways for rpm-serde to encode maps, this list of pairs is probably the best solution for compatibility.

Reply:
Oh yeah, with "array" I meant the `[u8; 32]`. That we can then use `Checksum` directly, not the byte array

Reply:
```suggestion
    // TODO: Remove the array usage as soon as `Checksum` has a stable wire format in msgpack
```

Oh, you mean like this?

Reply:
Yep!